[VEN-1548]: add explanation about the mechanism implemented in the Oracles (#88)
VenusProtocol/oracle: README.md

> # Resilient Oracle

Author: ```suggestion
# Resilient Price Feed
```

> Resilient Oracle is the price oracle used by the [Venus Protocol](https://app.venus.io) to fetch asset prices.

Author: ```suggestion
Resilient Price Feed is the logic used by the [Venus Protocol](https://app.venus.io) to fetch asset prices.
```

DeFi protocols are vulnerable to incorrectly reported prices which can lead to lost money. A price oracle can be manipulated,fail, or suffer other attacks depending on the type of price oracle and creates a single point of failure and opening attack vectors to the protocol if not mitigated.

> The Resilient Oracle uses multiple oracle sources and fallback mechanisms to return accurate prices and protect from oracle failures. Currently, it includes integrations with Chainlink, Pyth, Binance Oracle and TWAP (Time-Weighted Average Price) oracles. TWAP uses PancakeSwap as the on-chain price source.

Author: ```suggestion
The Resilient Price Feed uses multiple oracle sources and fallback mechanisms to return accurate prices and protect from oracle failures. Currently, it includes integrations with Chainlink, Pyth, Binance Oracle and TWAP (Time-Weighted Average Price) oracles. TWAP uses PancakeSwap as the on-chain price source.
```

## Details

> The Resilient Oracle configures a main, pivot and fallback oracle for every market (vToken). The main oracle oracle is the most trustworthy price source, the pivot oracle is used as a loose sanity checker and the fallback oracle is used as a backup price source.

Author: ```suggestion
The Resilient Oracle configures a main, pivot and fallback oracle for every market (vToken). The main oracle is the most trustworthy price source, the pivot oracle is used as a loose sanity checker and the fallback oracle is used as a backup price source.
```

> The Resilient Oracle configures a main, pivot and fallback oracle for every market (vToken). The main oracle oracle is the most trustworthy price source, the pivot oracle is used as a loose sanity checker and the fallback oracle is used as a backup price source.

Author: ```suggestion
The Resilient Price Feed configures a main, pivot and fallback oracle for every market (vToken). The main oracle oracle is the most trustworthy price source, the pivot oracle is used as a loose sanity checker and the fallback oracle is used as a backup price source.
```

> ![Resilient Oracle](./marketing-assets/oracles.png)

Author: ```suggestion
![Resilient Price Feed](./marketing-assets/oracles.png)
```

Oracle prices are validated using an **upper** and **lower** bound ratio, which is set for every market. The **upper bound ratio** represents the **deviation** between the **reported price** (price from oracle that’s being validated) and anchor price (price from oracle we are validating against) beyond which the reported price will be invalidated. The **lower bound ratio** presents the **deviation** between **reported price and anchor price** below which the reported price will be invalidated. For an oracle price to be considered valid the below statement should be true:

```
anchorRatio = anchorPrice/reporterPrice
isValid = anchorRatio <= upperBoundAnchorRatio && anchorRatio >= lowerBoundAnchorRatio
```

> The default configuration uses Chainlink as the main oracle, TWAP or Pyth oracle as pivot oracle depending on which supports the given market and Binance oracle as the fallback oracle. For some markets we may use Pyth or TWAP as the main oracle if the token price is not supported by Chainlink or Binance oracles.

Reviewer: TWAP is very inaccurate, I'd rater we not use it for anything other than pivot
